service/resource/endpoint/create.go

```diff
@@ -9,15 +9,15 @@ import (
 	apiv1 "k8s.io/client-go/pkg/api/v1"
 )
 
-func (r *Resource) GetCreateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+func (r *Resource) newCreateChange(ctx context.Context, obj, currentState, desiredState interface{}) (Endpoint, error) {
```

Comment:
The return value should remain `interface{}`.

Reply:
Done, I saw this in `azure-operator` I believe, so I wasn't sure.
Here to be exact: https://github.com/giantswarm/azure-operator/blob/cc829fc9665e8ef51ebe393061cfca8f56625a09/service/resource/deployment/resource.go#L269
So is this a preference or should we be consistent with this?

Reply:
Interesting. I have missed this in the azure-operator review. I asked Pawel. Lets see what we can come up with. Shouldn't block us now.